lib/Worker.js

```diff
@@ -53,8 +62,9 @@ module.exports = function(redis) {
 
 		// if we have a timout set, just clear it and return
 		if(self.timeout !== null){
```

Comment:
Great job catching the fact that the `setImmediate` ID was never set! This could have made for a really weird stopping case, if:

1. the process weren't shutdown after `stop()` was finished
2. the `stop()` call happened to be called in between `setImmediate` being called, and the end of the current processor tick

To finish your fix, I believe this line should be:
```js
if(self.timeout !== null || self.immediate !== null){
```

I believe my intent was just to use `self.timeout` for both the `setTimeout` ID and `setImmediate` ID, but your implementation is much saner.

Reply:
Can the clear* methods all be used interchangeably on the set* methods? Just testing in the terminal it looks they are interchangeable. I only separated them because I thought they needed to be called respectively.

Reply:
The IDs returned by the set* methods are opaque values (probably a number or string) and calling clear* with the wrong type won’t throw (it just returns false I believe). Honestly, the setImmediate spec is now considered deprecated (abandoned, really, although it will probably continue to be available in node for a long time). You could alternatively `setTimeout` with a value of zero instead, as we really don’t depend on `setImmediate`’s special “check” phase timing :)